### PR TITLE
[CI] Monorepo diff plugin for integration tests

### PR DESCRIPTION
**Pull request by pazone**
## What does this PR do?
Use monorepo diff plugin to run integration tests for pull requests and branches.
The plugin specifies a list of watched paths for pull requests. If a PR modifies files under a path from the list - integration tests will be executed. 

The main pipeline triggers `elastic-agent-extended-testing` if all conditions are met:
* At least one of the listed paths is modified by a PR
* `skip-it` label is not set

When a PR is merged to main or a release branch the integration tests will be triggered in any case

That's how it looks now:

Screenshot 2024-06-12 at 15 47 30

## Why is it important?
The [previous approach](https://github.com/elastic/elastic-agent/pull/4836) wasn't flexible enough and the separate integration testing pipeline didn't allow to run integration tests conditionally due to required checks limitation. 

**Important**: if we add `skip-it` label - the PR will be mergeable without running integration tests. 
## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

## Additional checks

- [ ] Draft PRs shouldn't trigger integration tests by default (Need to enhance [buildkite-pr-bot](https://github.com/elastic/buildkite-pr-bot) and pass `GITHUB_PULL_REQUEST_DRAFT` env variable to the build)
- [x] `/test` comment must trigger unit and integration tests also for draft PRs
- [x] A PR modifying the files from watch list must run integration tests
- [x] If a PR doesn't modify the listed files integration tests shouldn't be launched 
- [x] A PR labeled with `skip-it` must not trigger integration tests

## Related issues

-

## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

**Review by blakerouse — APPROVED** [review]
Looks good.

**Comment by pazone** [comment]
@cmacknz @blakerouse @rdner Could you please take a look at the [list of watched paths](https://github.com/elastic/elastic-agent/pull/4906/files#diff-a3991ebf1475eb82acab13946ffc1eca02b43917f6d5bec3898f45c8b0b9bd53R247)?

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -232,3 +232,48 @@ steps:
...
+                - internal/
+                - dev-tools/
+                - pkg/
+                - testing/
+                - .agent-versions.json
+                - .go-version
+                - .package-version
+                - go.mod
+                - magefile.go
+                - main.go
```
Comment (blakerouse) [inline .buildkite/pipeline.yml:256]: Missing `version/`, `specs/`, `go.sum`.

Comment (pazone) [reply .buildkite/pipeline.yml:256]: Thanks!

**Pull request merged**